tests/workflow_contracts/nixie_toolchain_test.py

```
@@ -0,0 +1,92 @@
+
+
+def _build_steps() -> list[dict[str, object]]:
+    """Return the steps from the CI build job."""
+    workflow = cast(
+        "object", yaml.safe_load(WORKFLOW_PATH.read_text(encoding="utf-8"))
+    )
+    match workflow:
+        case {"jobs": dict() as jobs}:
+            pass
+        case _:
+            raise AssertionError("the CI workflow must declare jobs")
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)